### PR TITLE
Update image test job

### PR DESCRIPTION
**Pull request by jhadvig**
Now the images are going to be build and tested with their STI test framework. Both CentOS7 and Rhel7 images are build and tested.

### CODE REVIEW

**Comment by jhadvig** [comment]
@mfojtik @danmcp PTAL

File: lib/vagrant-openshift/command/test_openshift3_image.rb [hunk 1]
```diff
@@ -103,31 +103,20 @@ def execute
...
+if [ #{image} != "sti-base" ]; then
+  # Pull base images
+  docker pull ci.dev.openshift.redhat.com:5000/openshift/base-centos7 && docker tag ci.dev.openshift.redhat.com:5000/openshift/base-centos7 openshift/base-centos7
+  docker pull ci.dev.openshift.redhat.com:5000/openshift/base-rhel7 && docker tag ci.dev.openshift.redhat.com:5000/openshift/base-rhel7 openshift/base-rhel7
+fi
 
-# Build the STI image we use for testing
-docker build -t #{image}-candidate .
+# Test CentOS 7 based images with STI test framework
+make test VERSION=#{image_version}
```
Comment (mfojtik) [inline lib/vagrant-openshift/command/test_openshift3_image.rb]: we said that the only convention would be to have 'test/run', having this in place changes that convention to have 'make test VERSION=??'  command...

File: lib/vagrant-openshift/command/test_openshift3_image.rb [hunk 2]
```diff
@@ -103,31 +103,20 @@ def execute
 # switch to the desired ref
 git checkout #{ref}
 
-# If software version is set, change directory to it
-[ -n "#{image_version}" ] && pushd #{image_version} > /dev/null
+if [ #{image} != "sti-base" ]; then
```
Comment (danmcp) [inline lib/vagrant-openshift/command/test_openshift3_image.rb]: Should put #{image} in quotes

File: lib/vagrant-openshift/command/test_openshift3_image.rb [hunk 3]
```diff
@@ -103,31 +103,20 @@ def execute
...
-  # If software version is set, go back to the repository root
-  [ -n "#{image_version}" ] && popd > /dev/null
-fi
-
-if [ $status -eq 0 ]; then
-  # get the image id
-  image_id=`docker inspect --format="{{ .Id }}" #{image}-candidate:latest`
-
-  # tag it devenv-ready
-  docker tag $image_id #{image}:devenv-ready
```
Comment (danmcp) [inline lib/vagrant-openshift/command/test_openshift3_image.rb]: Is devenv-ready really appropriate as a new term?  We have qe-ready.  Are these images targeted to be ready just for devenvs?

Comment (mfojtik) [inline lib/vagrant-openshift/command/test_openshift3_image.rb]: We should not do any tagging here. We are doing tagging here: 

https://ci.openshift.redhat.com/jenkins/job/push_openshift3_images

Also we don't tag 'devenv-ready' (neither 'qa-ready'), we just tag with the GIT commit short ref AND as latest. QA then can just simply pull the image to get the latest one...

**Comment by jhadvig** [comment]
I've added new flag `--base_images` which indicates whether the base-centos7 and base-rhel7 should be pulled from our registry for building STI based images.
We will have to update the format of the test_openshift3_image command
from:
`vagrant test-openshift3-image --image openshift/$JOB_NAME --ref $REF --image_version 2.0`

to:
`vagrant test-openshift3-image --image openshift/ruby --source openshift/sti-ruby --ref $REF --image_version 0.10 --base_image`

With this logic we will be able to build and test images, even if they dont have different versions. Also the Dockerfile.rhel7 will be optional.

**Comment by jhadvig** [comment]
I've tested it locally and everything worked as expected. The only thing that has to be done (after merge) is to update all the Jenskins job that are using this cmd.

**Pull request merged**